### PR TITLE
Reduce Layer documentation

### PR DESCRIPTION
**Pull request by Atticus1806**
The documentation of the reduce layer was outdated since more than just `sum` and `max` are allowed. Still kept them as examples, but maybe we can just drop the completely from there.

### CODE REVIEW

**Comment by Atticus1806** [comment]
What I primarily added was the "generalization". Because from reading: 
`This reduces some axis by using "sum" or "max".` I would stop reading and think I can only use these two modes.
We can also just do like 
```
This reduces some axis by using a method defined by the mode parameter
It's basically a wrapper around the corresponding tensorflow functions
```

**Comment by albertz** [comment]
As I said, I would keep it short, like: "This reduces some axis, e.g. by using "sum" or "max"."

**Comment by albertz** [comment]
Also, I would have expected that it's already clear from the layer name (`ReduceLayer`) that sum and max are only examples.

**Comment by albertz** [comment]
Btw, the documentation always anyway should also be read together with the arguments. When you look at the generated doc: https://returnn.readthedocs.io/en/latest/layer_reference/basic.html?highlight=reducelayer#returnn.tf.layers.basic.ReduceLayer

It already becomes pretty clear what it does. I wonder if adding more text really adds any value, or rather makes it just more convoluted. But this is just my opinion. Maybe someone else could also state his opinion.

**Comment by Atticus1806** [comment]
>"This reduces some axis, e.g. by using "sum" or "max"

I can live with that too. Just thought it would be helpful to make clear sum and max are just examples. 

> Also, I would have expected that it's already clear from the layer name (ReduceLayer) that sum and max are only examples.

I am not sure if I agree. Even though different reductions are possible its not always sure they are implemented (yet)

In the end this is really minor, just thought it would be helpful.
If we just add the 'e.g." then im fine with the second line staying the same.

**Comment by albertz** [comment]
> > "This reduces some axis, e.g. by using "sum" or "max"
> 
> I can live with that too. Just thought it would be helpful to make clear sum and max are just examples.

The "e.g." says exactly that, right?

> > Also, I would have expected that it's already clear from the layer name (ReduceLayer) that sum and max are only examples.
> 
> I am not sure if I agree. Even though different reductions are possible its not always sure they are implemented (yet)

But then this is not a problem of the documentation but about the implementation. If you find out that sth is missing, you should not edit the documentation but instead just add whatever is missing.

> In the end this is really minor, just thought it would be helpful. If we just add the 'e.g." then im fine with the second line staying as before.

Ok, at least I would prefer that.

**Review by albertz — APPROVED** [review]
